### ISSUE TITLE
sp-tab should support "href" attribute

### ISSUE THREAD

**Issue by lazd** [adobe/spectrum-web-components#608]
### Expected Behaviour

I want to be able to have Tabs that are used for navigation and consist of a number of links.

### Actual Behavior

The `href` attribute does nothing.

#### Platform and Version

`@spectrum-web-components/tab@0.3.6`
`@spectrum-web-components/tab-list@0.3.6`

#### Sample Code that illustrates the problem

```html
<sp-tab-list selected="1">
    <sp-tab href="/page1" label="Tab 1" value="1" tabindex="1"></sp-tab>
    <sp-tab href="/page2" label="Tab 2" value="2" tabindex="2"></sp-tab>
    <sp-tab href="/page3" label="Tab 3" value="3" tabindex="3"></sp-tab>
    <sp-tab href="/page4" label="Tab 4" value="4" tabindex="4"></sp-tab>
</sp-tab-list>
```

#### Context

These tabs are being used as top-level navigation for website. As such, I'll also need to apply `role="navigation"`, which I assume will work just fine on `<sp-tab-list>`, but would like some insight into whether that is the expected implementation.

![image](https://user-images.githubusercontent.com/201344/79006431-fd23e200-7b0d-11ea-8856-a61261a6e4cc.png)

**Comment by Westbrook** [comment]
@lazd This is a great request! I'd been thinking about similar as I investigate rewriting our documentation site as a static site generated by 11ty, but had mainly relied on this pattern in the context of an SPA-like page up until now. Currently the `sp-tab` element does not feature `href` in it's API: https://opensource.adobe.com/spectrum-web-components/components/tab/api Partially, this comes from the fact that the Spectrum CSS package supporting the delivery of this pattern does not have styles that would support for the `.spectrum-Tabs-itemLabel` element being an `<a>` element. Take a look at a manual application of such an element in the Spectrum CSS site as it is shipped today:

![image](https://user-images.githubusercontent.com/1156657/79007553-61a06a80-7b29-11ea-8613-fcd9ff3cd591.png)

Do you think that we could get the associated update outlined here into Spectrum CSS? If so, I'd be happy to add this feature!

On the question of applying `role=navigation` to `sp-tab-list`, I think that you would normally be covered in that case, but currently `sp-tab-list` forces the value of `role` to `tab-list` by default, so flexibility there would be part of the above additions.

@jnurthen can you back us up on whether the use of the `navigation` role in the context would be correct? Particularly, if `navigation` is acceptable, should `navigation` replace the current `role=tab-list` or can you have multiple roles? I look forward to being able to add this information to the documentation for this element so it's easier to deploy correctly going forward.

**Comment by jnurthen** [comment]
The tab pattern is for web application tabs not web page navigation tabs.
I'd actually suggest using a different component rather than trying to use the same one.

This navigation component would simply be a nav which contains links for each of the tabs. A user would tab through the page - there would be no arrow key behavior. This component would be very simple. 
The tab component should remain as it has its uses - but web page navigation is not one of them.

**Comment by Westbrook** [comment]
Thanks for the insight here @jnurthen!

@lazd, with this in mind, do you think that we'd be able to get the Spectrum CSS team to ship a "Top Nav" sort of patterns with the updated the `<label>` styling to manage an `<a>`? SWC could technically ship two different components from the same `@spectrum-web-components/tab` package (with the addition to the `<a>` styling) but with Jason's note above it probably is safe long term not to conflate the two patterns, even at the Spectrum CSS level.

Thoughts?

**Comment by lazd** [comment]
Thanks for investigating!

Hmm, it sounds like we need a way paired-down web component that looks like a Tablist, but doesn't have any arrow key or focus trapping behavior.

It seems like the easiest way to achieve that would be to make the Spectrum CSS component support `<a>` inside of it like you said [here](https://github.com/adobe/spectrum-web-components/issues/608#issuecomment-612122541), and Spectrum Web Components could use the component with a `<a>` for its `TopNav` component.

The other option is to go to design, ask if that component needs to be a separate thing, come up with a name, build it in Spectrum CSS, then build it in Spectrum Web Components. I know that, currently, they don't distinguish Tablists that do Tablist things from Tablists that do site navigation things, but they could be swayed.

**Comment by lazd** [comment]
@Westbrook support for anchors as labels implemented in https://github.com/adobe/spectrum-css/pull/672
